### PR TITLE
8257993: vmTestbase/nsk/jvmti/RedefineClasses/StressRedefine/TestDescription.java crash intermittently

### PR DESCRIPTION
**Pull request by RealLucy**
This backport is for parity with 11.0.21-oracle

Patch (from jdk16) did not apply at all. Basically, it is a fresh implementation, guided by the original fix. Had to introduce new method "Method* get_new_method() const;" in method.hpp. Did not exist in 11.

Local testing successful.
SAP internal nightly build and test did not show any issues.
Reviewed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8257993](https://bugs.openjdk.org/browse/JDK-8257993): vmTestbase/nsk/jvmti/RedefineClasses/StressRedefine/TestDescription.java crash intermittently (**Bug** - P3)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [25c47887](https://git.openjdk.org/jdk11u-dev/pull/2051/files/25c478875973774cc10ad85fe472d449f2feb2cc)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2051/head:pull/2051` \
`$ git checkout pull/2051`

Update a local copy of the PR: \
`$ git checkout pull/2051` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2051/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2051`

View PR using the GUI difftool: \
`$ git pr show -t 2051`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2051.diff">https://git.openjdk.org/jdk11u-dev/pull/2051.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2051#issuecomment-1644046509)

### CODE REVIEW

**Review by TheRealMDoerr — APPROVED** [review]
LGTM. Thanks for backporting!

File: src/hotspot/share/interpreter/interpreterRuntime.cpp [hunk 1]
```diff
@@ -854,24 +854,18 @@ void InterpreterRuntime::resolve_invoke(JavaThread* thread, Bytecodes::Code byte
...
-        // in the middle of resolve. If it is looping here more than 100 times
-        // means then there could be a bug here.
-        guarantee((retry_count++ < 100),
-                  "Could not resolve to latest version of redefined method");
-        // method is redefined in the middle of resolve so re-try.
-        LinkResolver::resolve_invoke(info, receiver, pool,
-                                     last_frame.get_index_u2_cpcache(bytecode), bytecode,
-                                     CHECK);
-      }
+
```
Comment (TheRealMDoerr) [inline src/hotspot/share/interpreter/interpreterRuntime.cpp:864]: Original change doesn't introduce this empty line. I suggest to remove it before integrating.

Comment (RealLucy) [reply src/hotspot/share/interpreter/interpreterRuntime.cpp:864]: Will do.
Thanks for the review!

**Pull request closed without merging**